Doc/c-api/init.rst

```diff
@@ -2226,6 +2226,12 @@ is resumed, and its locks reacquired.  This means the critical section API
 provides weaker guarantees than traditional locks -- they are useful because
 their behavior is similar to the :term:`GIL`.
 
+Critical sections are appropriate for protecting custom types defined by your
+C-API extensions. They should generally not be used with built-in types like
+:c:struct:`PyDictObject` or :c:struct:`PyListObject` because their public APIs
+already use critical sections internally, with the notable exception of
+:c:func:`PyDict_Next`.
```

Comment:
```suggestion
already use critical sections internally, with the :ref:`notable
exception<PyDict_Next>` of :c:func:`PyDict_Next`.
```

It looks like there's already a link anchor in the free-threading C extension howto guide that talks about this, so you can just use the cross-reference.